### PR TITLE
Optuna Callback

### PR DESCRIPTION
**Pull request by elephantmipt**
## Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contribution guide](https://github.com/catalyst-team/catalyst/blob/master/CONTRIBUTING.md)?
- [x] Did you check the code style? `catalyst-make-codestyle && catalyst-check-codestyle` (`pip install -U catalyst-codestyle`).
- [x] Did you make sure to update the docs? We use Google format for all the methods and classes.
- [x] Did you check the docs with `make check-docs`?
- [x] Did you write any new necessary tests?
- [x] Did you add your new functionality to the docs?
- [x] Did you update the [CHANGELOG](https://github.com/catalyst-team/catalyst/CHANGELOG.md)?
- [x] **You can use 'Login as guest' to see Teamcity build logs.**

## Type of Change

- [ ] Examples / docs / tutorials / contributors update
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves an existing feature)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

### CODE REVIEW

**Comment by elephantmipt** [comment]
tests will be soon

File: catalyst/contrib/dl/callbacks/optuna_callback.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
+import optuna
+
+from catalyst.core import Callback, CallbackOrder, IRunner
+
+
+class OptunaCallback(Callback):
```
Comment (Scitator) [inline catalyst/contrib/dl/callbacks/optuna_callback.py:6]: let's add it to the docs
https://github.com/catalyst-team/catalyst/blob/master/docs/api/contrib.rst

File: catalyst/contrib/dl/callbacks/optuna_callback.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        Considering prune or not to prune current run at current epoch.
+
+        Raises:
+            TrialPruned: if current run should be pruned
+
+        Args:
+            runner: runner for current experiment
+        """
+        metric_value = runner.valid_metrics[runner.main_metric]
+        self.trial.report(metric_value, step=runner.epoch)
```
Comment (Scitator) [inline catalyst/contrib/dl/callbacks/optuna_callback.py:65]: is it possible to report something more, that one metric only?

Comment (elephantmipt) [reply catalyst/contrib/dl/callbacks/optuna_callback.py:65]: Nope. Why do we need something more than metric?

**Pull request merged**